### PR TITLE
Add support for ed25519 based keys

### PR DESCRIPTION
**Pull request by priyawadhwa**
We can use this in tekton chains! 

Signed-off-by: Priya Wadhwa <priyawadhwa@google.com>

### CODE REVIEW

**Review by lukehinds — COMMENTED** [review]
~~how do you see this being called? I can't see any wiring up to the APIs. Is this instead something that will be imported into chains and used there?~~

forget the above, I thought I was in sigstore/fuclio

File: pkg/signature/ed25519.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+type ED25519SignerVerifier struct {
+	ED25519Verifier
+	Key ed25519.PrivateKey
+}
+
+func (s ED25519SignerVerifier) Sign(_ context.Context, rawPayload []byte) (signature, signed []byte, err error) {
+	signature = ed25519.Sign(s.Key, rawPayload)
+	return
+}
```
Comment (lukehinds) [inline pkg/signature/ed25519.go:40]: might be worth capturing an err here, such as:

https://github.com/sigstore/sigstore/blob/11e6eaab7cdc5cea7113004d2de344fe0f7fb640/pkg/signature/ecdsa.go#L45-L48

**Review by lukehinds — COMMENTED** [review]
The code looks good, but just thinking about the layout / organisation. 

We already have `/pkg/signature/ecdsa.go` could we not just pass in the curve type we want there and have all ecliptic handling in ecdsa.go 

e.g. use the below and instead of having `elliptic.P256()` set, we allow folks to pass in their own preferred code?

https://github.com/sigstore/sigstore/blob/11e6eaab7cdc5cea7113004d2de344fe0f7fb640/pkg/signature/ecdsa.go#L80-L86

File: pkg/signature/ed25519.go (hunk 1 above)
Comment (priyawadhwa) [reply pkg/signature/ed25519.go:40]: The [ed25519.Sign](https://golang.org/pkg/crypto/ed25519/#Sign) function actually only returns `[]byte` :)

**Comment by priyawadhwa** [comment]
>We already have /pkg/signature/ecdsa.go could we not just pass in the curve type we want there and have all ecliptic handling in ecdsa.go

IIUC ecdsa & ed25519 are different algorithms, so I don't think passing in a different curve to ecdsa would necessarily give you ed25519 (but I could be wrong about that)

**Comment by dlorenc** [comment]
> IIUC ecdsa & ed25519 are different algorithms, so I don't think passing in a different curve to ecdsa would necessarily give you ed25519 (but I could be wrong about that)

Yup, different curves.

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**